cmd/spire-agent/cli/run/run.go

```diff
@@ -248,9 +248,12 @@ func newAgentConfig(c *config) (*agent.Config, error) {
 	ac.DataDir = c.Agent.DataDir
 	ac.EnableSDS = c.Agent.EnableSDS
 
-	ll := strings.ToUpper(c.Agent.LogLevel)
-	lf := strings.ToUpper(c.Agent.LogFormat)
-	logger, err := log.NewLogger(ll, lf, c.Agent.LogFile)
+	logOptions = append(logOptions,
+		log.Level(c.Agent.LogLevel),
+		log.Format(c.Agent.LogFormat),
+		log.OutputFile(c.Agent.LogFile))
```

Comment:
Assuming I'm reading this correctly, the behavior looks like it should be "CLI flags and config file options override logOptions injected by main". Which means that injected log options of these types serve as default values only. I think that makes sense from an experience standpoint..?

At any rate, we should assert this behavior somewhere. I think the Run test suite can do it by examining the resulting logger.

Reply:
right, log options are applied in-order, so config file would take precedence.

I'll add a test verifying that behavior.

Reply:
added a test